app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt

```diff
@@ -545,11 +555,16 @@ open class DefaultToolbarMenu(
         R.drawable.ic_synced_tabs,
         primaryTextColor()
     ) {
-        onItemTapped.invoke(ToolbarMenu.Item.SyncAccount)
+        onItemTapped.invoke(Item.SyncAccount(signedInToFxa))
     }
 
     @VisibleForTesting(otherwise = PRIVATE)
     val newCoreMenuItems by lazy {
+
+        // Predicates that are called once, during screen init
+        val shouldShowSaveToCollection = (context.asActivity() as? HomeActivity)
```

Comment:
I'm not sure what this is related to and referencing with the bug for #19114 . Let's remove it if it's not needed or file a separate issue for it and reference it to a new PR?